test/SlowTests/Server/Replication/PullReplicationPreventDeletionsTests.cs

```diff
@@ -0,0 +1,395 @@
+                HubName = "pullRepHub"
+            }));
+
+            using (var s = sinkStore.OpenAsyncSession())
+            {
+
+                dynamic user1 = new {Source = "Sink"};
+                await s.StoreAsync(user1, "users/insink/1");
+                s.Advanced.GetMetadataFor(user1)[Constants.Documents.Metadata.Expires] = DateTime.UtcNow.AddMinutes(10);
+
+                dynamic user2 = new {Source = "Sink"};
+                await s.StoreAsync(user2, "users/insink/2");
+                s.Advanced.GetMetadataFor(user2)[Constants.Documents.Metadata.Expires] = DateTime.UtcNow.AddMinutes(10);
+
+                await s.SaveChangesAsync();
+            }
+
+            using (var s = hubStore.OpenAsyncSession())
+            {
+                await s.StoreAsync(new {Source = "Hub"}, "users/inhub/1");
+                await s.SaveChangesAsync();
+            }
+
+            var sinkDatabaseInstance = await GetDocumentDatabaseInstanceFor(sinkStore);
+            sinkDatabaseInstance.ForTestingPurposes.ForceSendTombstones = true;
+
+            await EnsureReplicatingAsync(hubStore, sinkStore);
+            await EnsureReplicatingAsync(sinkStore, hubStore);
+
+            using (var h = hubStore.OpenAsyncSession())
+            {
+                //check hub got both docs
+                var doc1 = await h.LoadAsync<dynamic>("users/insink/1");
+                Assert.NotNull(doc1);
+
+                var doc2 = await h.LoadAsync<dynamic>("users/insink/2");
+                Assert.NotNull(doc2);
+
+                //check expired does not exist in users/insink/1
+                var metadata = h.Advanced.GetMetadataFor(doc1);
+                var errorMD = Assert.Throws<KeyNotFoundException>(() => metadata[Constants.Documents.Metadata.Expires]);
+                Assert.Contains("@expires is not in the metadata", errorMD.Message);
+
+                //check expired does not exist in users/insink/2
+                metadata = h.Advanced.GetMetadataFor(doc2);
+                errorMD = Assert.Throws<KeyNotFoundException>(() => metadata[Constants.Documents.Metadata.Expires]);
+                Assert.Contains("@expires is not in the metadata", errorMD.Message);
+            }
+
+            Assert.True(WaitForDocument(sinkStore, "users/inhub/1"));
+
+            var hubDatabaseInstance = await GetDocumentDatabaseInstanceFor(hubStore);
+            var error = "";
+            hubDatabaseInstance.ReplicationLoader.IncomingHandlers.ToArray()[0].Failed += (handler, exception) =>
+            {
+                error = exception.Message;
+            };
+
+            //delete doc from sink
+            using (var s = sinkStore.OpenAsyncSession())
+            {
+                s.Delete("users/insink/1");
+                await s.SaveChangesAsync();
+            }
+
+            //make sure doc deleted from sink
+            Assert.False(WaitForDocument(sinkStore, "users/insink/1"));
```

Comment:
That is probably wrong. you are using this as a timeout, but it shouldn't be the case.
Instead, wait for the `Failed` event to be raised.